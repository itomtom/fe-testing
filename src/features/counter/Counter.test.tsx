/**
 * @jest-environment jsdom
 */

import React from "react";
import { Provider } from "react-redux";
import { Store } from "redux";

import { render, screen } from "@testing-library/react";
import { Counter } from "./Counter";
import { CounterState } from "./counterSlice";
import counterReducer from "./counterSlice";
import { configureStore } from "@reduxjs/toolkit";

const initialState: CounterState = {
  value: 3,
  status: "idle",
};

const getStore = (state = initialState) =>
  configureStore({
    preloadedState: {
      counter: state,
    },
    reducer: {
      counter: counterReducer,
    },
  });

const renderer = (store = getStore()) => {
  render(
    <Provider store={store}>
      <div data-testid="component-root">
        <Counter />
      </div>
    </Provider>
  );
};

describe("Counter", () => {
  it("should show join button", () => {
    const store = getStore(initialState);
    renderer(store);
    expect(screen.getByTestId("component-root")).toMatchSnapshot();
  });
});
