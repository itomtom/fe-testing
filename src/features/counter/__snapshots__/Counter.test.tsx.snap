// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter should show join button 1`] = `
<div
  data-testid="component-root"
>
  <div>
    <div
      class="row"
    >
      <button
        aria-label="Decrement value"
        class="button"
      >
        -
      </button>
      <span
        class="value"
      >
        3
      </span>
      <button
        aria-label="Increment value"
        class="button"
      >
        +
      </button>
    </div>
    <div
      class="row"
    >
      <input
        aria-label="Set increment amount"
        class="textbox"
        value="2"
      />
      <button
        class="button"
      >
        Add Amount
      </button>
      <button
        class="asyncButton"
      >
        Add Async
      </button>
      <button
        class="button"
      >
        Add If Odd
      </button>
    </div>
  </div>
</div>
`;
